java/client/src/main/java/glide/api/logging/Logger.java

```diff
@@ -0,0 +1,192 @@
+     * Initialize a logger if it wasn't initialized before - this method is meant to be used when
+     * there is no intention to replace an existing logger. The logger will filter all logs with a
+     * level lower than the given level. If given a <code>fileName</code> argument, will write the
+     * logs to files postfixed with <code>fileName</code>. If <code>fileName</code> isn't provided,
+     * the logs will be written to the console.
+     *
+     * @param level Set the logger level to one of <code>
+     *     [DISABLED, DEFAULT, ERROR, WARN, INFO, DEBUG, TRACE]
+     *     </code>. If log level isn't provided, the logger will be configured with default
+     *     configuration decided by Glide core.
+     * @param fileName If provided, the target of the logs will be the file mentioned. Otherwise, logs
+     *     will be printed to the console.
+     */
+    public static void init(@NonNull Level level, String fileName) {
+        if (loggerLevel == null) {
+            initLogger(level, fileName);
+        }
+    }
+
+    /**
+     * Initialize a logger if it wasn't initialized before - this method is meant to be used when
+     * there is no intention to replace an existing logger. The logger will filter all logs with a
+     * level lower than the default level decided by Glide core. Given a <code>fileName</code>
+     * argument, will write the logs to files postfixed with <code>fileName</code>.
+     *
+     * @param fileName The target of the logs will be the file mentioned.
+     */
+    public static void init(@NonNull String fileName) {
+        init(Level.DEFAULT, fileName);
+    }
+
+    /**
+     * Initialize a logger if it wasn't initialized before - this method is meant to be used when
+     * there is no intention to replace an existing logger. The logger will filter all logs with a
+     * level lower than the default level decided by Glide core. The logs will be written to stdout.
+     */
+    public static void init() {
+        init(Level.DEFAULT, null);
+    }
+
+    /**
+     * Initialize a logger if it wasn't initialized before - this method is meant to be used when
+     * there is no intention to replace an existing logger. The logger will filter all logs with a
+     * level lower than the given level. The logs will be written to stdout.
+     *
+     * @param level Set the logger level to one of <code>[DEFAULT, ERROR, WARN, INFO, DEBUG, TRACE]
+     *     </code>. If log level isn't provided, the logger will be configured with default
+     *     configuration decided by Glide core.
+     */
+    public static void init(@NonNull Level level) {
+        init(level, null);
+    }
+
+    /**
+     * Logs the provided message if the provided log level is lower then the logger level.
+     *
+     * @param level The log level of the provided message.
+     * @param logIdentifier The log identifier should give the log a context.
+     * @param message The message to log.
+     */
+    public static void log(
```

Comment:
Should we use Supplier<String> for message so that construction of the message doesn't occur if logging is below the logger's level?

Reply:
I suppose we could do this, yeah. It'd probably be worth updating Python and Node too in a separate PR.

Reply:
I realize probably want both a String log function and a Supplier one. There's some overhead when you use a lambda capturing local variables. So the caller should use the regular string overload when the String is already built or cheap to build, otherwise use the supplier